autopkg/autopkg_tools.py

```diff
@@ -276,15 +342,29 @@ def main():
     global DEBUG
     DEBUG = bool(opts.debug)
 
-    if not opts.list:
-        print("Recipe list not provided!")
-        sys.exit(1)
-
-    no_updates = []
+    failures = []
 
-    for recipe in parse_recipes(opts.list):
-        result = handle_recipe(recipe)
+    recipes = RECIPE_TO_RUN.split(", ") if RECIPE_TO_RUN else opts.list if opts.list else None
+    if recipes is None:
+        print("Recipe --list or RECIPE_TO_RUN not provided!")
+        sys.exit(1)
+    recipes = parse_recipes(recipes)
+    for recipe in recipes:
+        handle_recipe(recipe, opts)
         slack_alert(recipe, opts)
+        if not opts.disable_verification:
+            if not recipe.verified:
+                failures.append(recipe)
+    if not opts.disable_verification:
+        if failures:
+            title_file=open("pull_request_title","a+")
```

Comment:
Can we use context managers for cleaner file handling/implicit cleanup. eg:

```
with open("pull_request_title", "w+) as f:
    f.write()
```

Reply:
Just pushed the change.